src/OrchardCore.Modules/OrchardCore.PublishLater/Migrations.cs

```diff
@@ -1,59 +1,87 @@
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(ContentItemIndex.DocumentId)}",
+                "Id",
+                nameof(ContentItemIndex.DocumentId),
+                nameof(PublishLaterPartIndex.ContentItemId),
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc),
+                nameof(PublishLaterPartIndex.Published),
+                nameof(PublishLaterPartIndex.Latest))
+        );
+
+        return 3;
+    }
+
+    // This code can be removed in a later version.
+    public int UpdateFrom1()
+    {
+        // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
+        // this because dropping an index and altering a column don't work on all providers.
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .AddColumn<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
+            .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
+                nameof(ContentItemIndex.DocumentId),
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
+        );
+
+        return 2;
+    }
+
+    public int UpdateFrom2()
+    {
+        SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table =>
         {
-            // The 'ScheduledPublishUtc' column and related index are kept on existing databases,
-            // this because dropping an index and altering a column don't work on all providers.
+            table.AddColumn<string>(nameof(PublishLaterPartIndex.ContentItemId));
+            table.AddColumn<bool>(nameof(PublishLaterPartIndex.Published));
+            table.AddColumn<bool>(nameof(PublishLaterPartIndex.Latest));
 
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .AddColumn<DateTime>(nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
+            table.DropIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}");
 
-            SchemaBuilder.AlterIndexTable<PublishLaterPartIndex>(table => table
-                .CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc)}",
-                    "DocumentId",
-                    nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc))
-            );
+            table.CreateIndex($"IDX_{nameof(PublishLaterPartIndex)}_{nameof(ContentItemIndex.DocumentId)}",
+                "Id",
+                nameof(ContentItemIndex.DocumentId),
+                nameof(PublishLaterPartIndex.ContentItemId),
+                nameof(PublishLaterPartIndex.ScheduledPublishDateTimeUtc),
+                nameof(PublishLaterPartIndex.Published),
+                nameof(PublishLaterPartIndex.Latest));
+        });
```

Comment:
@hishamco I will approve but first did you try this migration step.

I mean do a setup before the code changes made in this PR, then check that the feature works, then update the code as done in this PR, ensure  that this migration step is run and works, and then that the feature still works.